core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -598,9 +618,9 @@ public String toString() {
         }
     }
 
-    private static String getCacheKey(ScriptEngineService scriptEngineService, String script) {
+    private static String getCacheKey(ScriptEngineService scriptEngineService, String name, String code) {
         String lang = scriptEngineService.types()[0];
-        return lang + ":" + script;
+        return lang + ":" + name + (code != null ? ":" + code : "");
```

Comment:
While this shouldn't affect uniqueness, to avoid confusion I will make it so inline scripts end up being lang:code.